core/exceptions/ExceptionStackTraceDisplay.php

```diff
@@ -68,73 +68,75 @@ protected function displayException(Exception $exception)
             ? $this->developerError($exception, $msg, $code, $trace_details)
             : $this->genericError($msg, $code);
         // start gathering output
-        $output = '
-<div id="ee-error-message" class="error">
-    ' . $error_message . '
-</div>';
-        $styles = $this->exceptionStyles();
+        $output = '<div id="ee-error-message" class="error">
+            ' . $error_message . '
+        </div>';
         $scripts = $this->printScripts(true);
         if (defined('DOING_AJAX')) {
-            echo wp_json_encode(array('error' => $styles . $output . $scripts));
+            echo wp_json_encode(array('error' => $output . $scripts));
```

Comment:
this might be an issue.

you'll need to try and chase down a spot in the code where exceptions are being caught and returned via AJAX and see what things look like.

I can't remember where that might be happening (EE core alone is over 620k lines of code now 😬) but I suspect it is during the registration process since that relies heavily on AJAX. You should also checkout and activate [MER (Multi Event Registration)](https://github.com/eventespresso/ee4-mer) as it also relies heavily on AJAX.

You'll then need to find a spot in the code where `ExceptionStackTraceDisplay` is being used and try to simulate an error state that will trigger this code during an AJAX response. Fun times 😆 

I'm totally open to suggestions as to how to solve this problem, but first you will need to understand the full scope of it.

Reply:
I simulated the AJAX error on normal registration and also MER using following code and the error with scripts and link tag (style) returns as response of AJAX request but it didn't show to the end user.

```
add_action('init', function()
{
    new \EventEspresso\core\exceptions\ExceptionStackTraceDisplay(new Exception('Simple Error'));
});
```

Please check this image.
<img width="1047" alt="Screen Shot 2022-05-26 at 20 56 57" src="https://user-images.githubusercontent.com/29144542/170540013-a05b479c-f61d-4dee-a85a-2c0797e557a8.png">

Reply:
can you plz switch to master branch, try your snippet there and see what happens?

if it doesn't display the error, then you will need to back trace the code, or find some other way to determine where those AJAX errors were being used, and find a way to reliably reproduce them. THEN switch back to this branch and test out your changes.

Reply:
I switched to **master** branch and added the previous code snippet in espresso.php file (main plugin file) and check the MER AJAX but it didn't show to user in frontend.

Then I added following codes in **EED_Multi_Event_Registration::remove_ticket** and some other functions. The errors was reflected in the AJAX response but they didn't show to user in frontend.

```
new \EventEspresso\core\exceptions\ExceptionStackTraceDisplay(new Exception('Simple Error'));
```

```
EE_Error::add_error(
    esc_html__( 'An invalid event object was received.', 'event_espresso' ),
    __FILE__,
    __FUNCTION__,
    __LINE__
);
```
<img width="1265" alt="Screen Shot 2022-05-30 at 18 08 29" src="https://user-images.githubusercontent.com/29144542/171005099-d8b1bb08-b12c-415f-92c7-da474fd228a3.png">

Also added the above codes into **EED_Single_Page_Checkout::process_ajax_request** function without MER and same happened.

<img width="1281" alt="Screen Shot 2022-05-30 at 18 23 15" src="https://user-images.githubusercontent.com/29144542/171007001-b7c7e25e-4ac1-4237-a9dc-9508fc64d52e.png">
<img width="1257" alt="Screen Shot 2022-05-30 at 18 24 18" src="https://user-images.githubusercontent.com/29144542/171007022-edc5957d-c606-4b92-b352-04ce55e4a797.png">

If the error display was working fine before then perhaps a recent change (most probably on JS codes) destroyed the functionality of error display. I can check it if I'm in the right path of investigation.

Reply:
Finally found fixed it. Please check the latest commit.

<img width="1243" alt="Screen Shot 2022-05-30 at 20 28 20" src="https://user-images.githubusercontent.com/29144542/171027987-ed41be93-d3f0-4990-926e-29f19bb84ad1.png">

Reply:
AWESOME! 
good job working through that.
sometimes it can take a long time to wade through all of the mountains of code to find the source of the issue, so thanks for sticking with it 💯